### PR TITLE
Use POSIX shared memory instead of SYSV shared memory on Linux and macOS

### PR DESCRIPTION
**Pull request by omichel**
By default Qt uses SYSV shared memory on Linux and macOS. This has several drawbacks:
- on macOS: the default size of SYSV shared memory segments is very small preventing many camera-based simulation to work out-of-the-box. We ask users to modify their system settings to work around this problem, which is not very user friendly.
- on Linux: the SYSV shared memory segments are not compatible with snap strict confinement, hence preventing us to package Webots as a snap package with strict confinement (which is better than the classic confinement).

### CODE REVIEW

**Comment by omichel** [comment]
:+1:

**Comment by fabienrohrer** [comment]
👀

**Comment by fabienrohrer** [comment]
Unfortunately, every camera are failing systematically, even if I clean the shared memory before (`./src/bin/clean_shm.sh`).

Capture d’écran 2019-09-10 à 09 51 13

Do you want to go ahead with this on Mac?

File: docs/reference/changelog-r2019.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Released on XXX YYth, 2019.
     - Added a `village_center` world.
   - Enhancements
     - Improved the intensity and color of the bus, truck and car vehicle lights.
+    - macOS and Linux: Use POSIX shared memory segments to fix the limit problems on macOS and allow for snap packaging with strict confinement on Linux.
```
Comment (fabienrohrer) [inline docs/reference/changelog-r2019.md]: I'm wondering if mentioning here `snap` is useful, it seems to be unrelated.

**Comment by omichel** [comment]
Oops. I tested only on Linux and it works fine. Unfortunately, I don't have a development environment ready on macOS, so it may take a while for me test investigate on macOS. Can I let you check it? I believe I didn't change anything from #855 which you tested successfully.

File: docs/reference/changelog-r2019.md (hunk 1 above)
Comment (omichel) [reply docs/reference/changelog-r2019.md]: It's related. Without POSIX shared memory, there is no way to create a strictly confined snap.

**Comment by fabienrohrer** [comment]
Just fixed an issue on macOS. Everything seems to work well now.

**Comment by omichel** [comment]
👍
